integration-cli/docker_cli_build_test.go

```diff
@@ -4526,85 +4527,87 @@ func TestBuildRenamedDockerfile(t *testing.T) {
-
 	if err != nil {
 		t.Fatal(err)
 	}
 	if !strings.Contains(out, "from dFile") {
-		t.Fatalf("Should have used dFile, output:%s", out)
+		t.Fatalf("test4 should have used dFile, output:%s", out)
 	}
 
 	out, _, err = dockerCmdInDir(t, ctx.Dir, "build", "--file=/etc/passwd", "-t", "test5", ".")
-
 	if err == nil {
-		t.Fatalf("Was supposed to fail to find passwd")
+		t.Fatalf("test5 was supposed to fail to find passwd")
 	}
-
 	if !strings.Contains(out, "The Dockerfile (/etc/passwd) must be within the build context (.)") {
-		t.Fatalf("Wrong error message for passwd:%v", out)
+		t.Fatalf("test5 - wrong error message for passwd:%v", out)
 	}
 
-	out, _, err = dockerCmdInDir(t, ctx.Dir+"/files", "build", "-f", "../Dockerfile", "-t", "test5", "..")
-
+	out, _, err = dockerCmdInDir(t, ctx.Dir+"/files", "build", "-f", "../Dockerfile", "-t", "test6", "..")
 	if err != nil {
-		t.Fatal(err)
+		t.Fatalf("test6 failed: %s", err)
 	}
-
 	if !strings.Contains(out, "from Dockerfile") {
-		t.Fatalf("Should have used root Dockerfile, output:%s", out)
+		t.Fatalf("test6 should have used root Dockerfile, output:%s", out)
 	}
 
-	out, _, err = dockerCmdInDir(t, ctx.Dir+"/files", "build", "-f", ctx.Dir+"/files/Dockerfile", "-t", "test6", "..")
-
+	out, _, err = dockerCmdInDir(t, ctx.Dir+"/files", "build", "-f", ctx.Dir+"/files/Dockerfile", "-t", "test7", "..")
 	if err != nil {
-		t.Fatal(err)
+		t.Fatalf("test7 failed: %s", err)
 	}
-
 	if !strings.Contains(out, "from files/Dockerfile") {
-		t.Fatalf("Should have used files Dockerfile - 2, output:%s", out)
+		t.Fatalf("test7 should have used files Dockerfile, output:%s", out)
 	}
 
-	out, _, err = dockerCmdInDir(t, ctx.Dir+"/files", "build", "-f", "../Dockerfile", "-t", "test7", ".")
-
+	out, _, err = dockerCmdInDir(t, ctx.Dir+"/files", "build", "-f", "../Dockerfile", "-t", "test8", ".")
 	if err == nil || !strings.Contains(out, "must be within the build context") {
-		t.Fatalf("Should have failed with Dockerfile out of context")
+		t.Fatalf("test8 should have failed with Dockerfile out of context: %s", err)
 	}
 
-	out, _, err = dockerCmdInDir(t, "/tmp", "build", "-t", "test6", ctx.Dir)
+	tmpDir := os.TempDir()
 
+	out, _, err = dockerCmdInDir(t, tmpDir, "build", "-t", "test9", ctx.Dir)
 	if err != nil {
-		t.Fatal(err)
+		t.Fatalf("test9 - failed: %s", err)
 	}
-
 	if !strings.Contains(out, "from Dockerfile") {
-		t.Fatalf("Should have used root Dockerfile, output:%s", out)
+		t.Fatalf("test9 should have used root Dockerfile, output:%s", out)
+	}
+
+	out, _, err = dockerCmdInDir(t, ctx.Dir+"/files", "build", "-f", "dFile2", "-t", "test10", ".")
+	if err != nil {
+		t.Fatalf("test10 should have worked: %s", err)
+	}
+	if !strings.Contains(out, "from files/dFile2") {
+		t.Fatalf("test10 should have used files/dFile2, output:%s", out)
+	}
+
+	out, _, err = dockerCmdInDir(t, tmpDir, "build", "-f", "Dockerfile", "-t", "test11", ctx.Dir)
```

Comment:
At `// Path to Dockerfile should be resolved relative to working directory, not relative to context.`